magenta/reviews/GAN.md

```diff
@@ -0,0 +1,134 @@
+
+Generative Adversarial Networks (GAN) is a framework for generating realistic samples from random noise first
+introduced by Goodfellow et al in 2014. At a high level, a GAN works by staging a battle between two neural networks.
+The first one generates samples. In parallel, the second tries to discriminate those samples from samples from
+a real dataset.
+We’ll first describe why we would want to do this and give some intuition behind what inspired GANs.
+We’ll then describe how they work, detail some disadvantages, and close with recent and future directions.
+
+<p align="center">
+  <img src="assets/gan/image00.gif"><br>
+  <i>Here are a set of faces generated purely from random noise.</i><br>
+  <i>Source: http://torch.ch/blog/2015/11/13/gan.html.</i>
+</p>
+
+So far deep generative models have had much less success than deep discriminative models.
+This is partly due to difficulty in approximating intractable probabilistic posteriors that arise in their
+optimization algorithms, which forces practitioners to use techniques like MCMC.
+It’s also because many of the methods that help train discriminative models do not work as well when applied to
+generative models.
+The adversarial framework in this paper can be trained purely with back propagation and helps solve some of
+these problems.
+GANs are an idea rooted in Game Theory.
+Two models compete against each other.
+The first, the generator, generates examples that it thinks are real.
+The second, the discriminator, tries to discriminate between examples from the dataset and examples drawn
+from the pool generated by the generator.
+The discriminator is trained in a supervised manner with two labels -- real or generated.
+The generator is trained to maximize the discriminator’s error on generated examples.
+The result is that it learns how to generate examples that approximate the real data distribution.
+With GANs, we approximate it by optimizing an adversarial loss:
+
+<p align="center">
+  <img align="center" src="assets/gan/image07.png"><br>
+</p>
+
+Those familiar with game theory might appreciate the insight that the optimization of this objective is like finding
+a Nash equilibrium of a minimax game between the generator and the discriminator.
+The loss has two components.
+The first one, <img src="assets/gan/image06.png">, makes the discriminator <img src="assets/gan/image01.png"> better at
+discriminating real examples from fake ones, and the second one, <img src="assets/gan/image13.png">, makes the generator
+<img src="assets/gan/image14.png"> generate samples that the discriminator considers real.
+The input, <img src="assets/gan/image02.png">, is a noise vector with distribution <img src="assets/gan/image10.png">.
+This is usually an N-dimensional normal or uniform distribution and serves as input to the network.
+
+Both parts can be independently optimized with gradient based optimization in turns.
```

Comment:
"independently optimized with gradient based optimization in turns" --> "independently optimized with gradient based methods".